eclipse-scout-core/src/desktop/outline/SearchOutline.ts

```diff
@@ -186,4 +270,133 @@ export class SearchOutline extends Outline implements SearchOutlineModel {
+    searchStates ||= new Set();
+
+    const searchStatesToAdd = new Set<SearchState>();
+    for (const searchState of searchStates) {
+      if (this._searchStates.has(searchState)) {
+        continue;
+      }
+      searchStatesToAdd.add(searchState);
+    }
+
+    const searchStatesToRemove = new Set<SearchState>();
+    for (const searchState of this._searchStates) {
+      if (searchStates.has(searchState)) {
+        continue;
+      }
+      searchStatesToRemove.add(searchState);
+    }
+
+    searchStatesToAdd.forEach(searchState => this._addSearchState(searchState));
+    searchStatesToRemove.forEach(searchState => this._removeSearchState(searchState));
+  }
+
+  protected _addSearchState(searchState: SearchState) {
+    if (!searchState) {
+      return;
+    }
+
+    if (this._searchStates.has(searchState)) {
+      return;
+    }
+    this._searchStates.add(searchState);
+
+    this._installSearchStateListeners(searchState);
+    this._updateSearchStatus();
+  }
+
+  protected _removeSearchState(searchState: SearchState) {
+    if (!searchState) {
+      return;
+    }
+
+    if (!this._searchStates.has(searchState)) {
+      return;
+    }
+    this._searchStates.delete(searchState);
+
+    this._uninstallSearchStateListeners(searchState);
+    this._updateSearchStatus();
+  }
+
+  protected _installSearchStateListeners(searchState: SearchState) {
+    if (!searchState) {
+      return;
+    }
+    searchState.on('propertyChange:resultCount', this._searchStateResultCountChangedHandler);
+    searchState.on('propertyChange:limited', this._searchStateLimitedChangedHandler);
+    searchState.on('propertyChange:pending', this._searchStatePendingChangedHandler);
+    searchState.one('destroy', this._searchStateDestroyHandler);
+  }
+
+  protected _uninstallSearchStateListeners(searchState: SearchState) {
+    if (!searchState) {
+      return;
+    }
+    searchState.off('propertyChange:resultCount', this._searchStateResultCountChangedHandler);
+    searchState.off('propertyChange:limited', this._searchStateLimitedChangedHandler);
+    searchState.off('propertyChange:pending', this._searchStatePendingChangedHandler);
+    searchState.off('destroy', this._searchStateDestroyHandler);
+  }
+
+  protected _onSearchStateResultCountChanged(event: PropertyChangeEvent<number>) {
+    this._updateSearchStatus();
+  }
+
+  protected _onSearchStateLimitedChanged(event: PropertyChangeEvent<boolean>) {
+    this._updateSearchStatus();
+  }
+
+  protected _onSearchStatePendingChanged(event: PropertyChangeEvent<boolean>) {
+    this._updateSearchStatus();
+  }
+
+  protected _onSearchStateDestroy(event: Event<SearchState>) {
+    this._removeSearchState(event.source);
+  }
+
+  get resultCount(): number {
+    let resultCount = 0;
+    for (const searchState of this._searchStates) {
+      resultCount += searchState.resultCount;
+    }
+    return resultCount;
+  }
+
+  get limited(): boolean {
+    for (const searchState of this._searchStates) {
```

Comment:
Or if you prefer it more: return Array.from(this._searchStates).some(state => state.limited); For me, both is fine